File: codewars/6kyu/Duplicate Encoder.py
from collections import Counter


def duplicate_encode(word):
    word = word.lower()
    w_list = [x for x in word]
    w_cnt = Counter(w_list)
    print(w_cnt)
    result = "".join([")" if w_cnt[x] != 1 else "(" for x in w_list])

    return result

import codewars_test as test

@test.describe("Duplicate Encoder")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(duplicate_encode("din"), "(((")
        test.assert_equals(duplicate_encode("recede"), "()()()")
        test.assert_equals(duplicate_encode("Success"), ")())())", "should ignore case")
        test.assert_equals(duplicate_encode("(( @"), "))((")

File: codewars/6kyu/Duplicate Encoder.js
function duplicateEncode(str) {
    const lowerStr = str.toLowerCase();
    const charCount = {};

    for (const char of lowerStr) {
        if (charCount[char] === undefined) {
            charCount[char] = 0;
        }
        charCount[char]++;
    }

    let result = '';
    for (const char of lowerStr) {
        if (charCount[char] > 1) {
            result += ')';
        } else {
            result += '(';
        }
    }

    return result;
}

console.log(duplicateEncode('din'))
console.log(duplicateEncode('recede'))
console.log(duplicateEncode('Success'))